src/containers/quiz/question-answer/Answer.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  FormControlLabel,
+  FormControlLabelProps,
+  Box,
+  Checkbox,
+  Radio,
+  InputAdornment,
+  SxProps
+} from '@mui/material'
+import { ChangeEventHandler, FC } from 'react'
+import CheckIcon from '@mui/icons-material/Check'
+import CloseIcon from '@mui/icons-material/Close'
+import { spliceSx } from '~/utils/helper-functions'
+import { styles } from './Answer.styles'
+import { QuestionTypesEnum } from '~/types'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import { questionType } from '~/components/question-editor/QuestionEditor.constants'
+import { useTranslation } from 'react-i18next'
+
+interface Answer {
+  text: string
+  isCorrect?: boolean
+  value?: string
+  label?: string
+  checked?: boolean
+  showCorrectness?: boolean
+  type: QuestionTypesEnum
+  isEditable?: boolean
+  sx?: SxProps
+  onTextInputChange?: ChangeEventHandler<HTMLInputElement>
+  onCheckboxChange?: (isChecked: boolean) => void
+}
+
+const Answer: FC<Answer> = ({
+  text,
+  isCorrect,
+  checked,
+  type,
+  label,
+  sx,
+  onTextInputChange,
+  onCheckboxChange,
+  value = '',
+  showCorrectness = false,
+  isEditable = true
+}) => {
+  const { isMultipleChoice, isOpenAnswer } = questionType(type)
+
+  const { t } = useTranslation()
+
+  const shouldShowState = (isOpenAnswer || checked) && showCorrectness
```

Comment:
Please think up of better name of this variable, because I did not understand what it does from the first glance

Reply:
There are many conditions so naming was the hard part) I made them more descriptive and replaced `state` by `status` here and in other places, because it was really confusing. Now this part looks like this: 
```
  const shouldShowAnswerStatus = (isOpenAnswer || checked) && showCorrectness
  const answerCorrectnessStatus = isCorrect
    ? AnswerStatusEnum.Correct
    : AnswerStatusEnum.Incorrect

  const answerStatus = shouldShowAnswerStatus
    ? answerCorrectnessStatus
    : AnswerStatusEnum.Unanswered
```